Week3/homework/2-step3.js

```diff
@@ -3,6 +3,28 @@
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const values = [];
   // Add your code here
+  for (let i = startIndex; i <= stopIndex; i++) {
+    values.push(i);
+    if (i % 3 === 0) {
+      threeCallback();
+    }
+    if (i % 5 === 0) {
+      fiveCallback();
+    }
+    if (i % 3 === 0 && i % 5 === 0)
+      threeCallback();
+    fiveCallback();
```

Comment:
You have not created a block with `{}` here.  as a result this last statement actually says:
```
if (i % 3 === 0 && i % 5 === 0) {
  threeCallback();
} else {
  fiveCallback();
}
```
So any value of `i` that is not divisible by `3` & `5` then it calls `fiveCallback`.
Coupled with the separate if statements, it means `fiveCallback` gets called far mor than it should have been.

I have run this code, and could tell immediately that something is wrong, because only 3 of the numbers are divisible by `3` or `5`.
Please *run* and *test* your code.  You could have figured out how many times you should have been seeing each message by hand, and then run the program to see if it matched your expectations.
Testing and running code is one of the most important parts of being a programmer.  Learn to love it.